### PR TITLE
fix: Remarks grouping in TextMate

### PR DESCRIPTION
**Pull request by jirimosinger**

### CODE REVIEW

File: clients/vscode-hlasmplugin/src/test/syntaxes/endevor/basics.listing.snap [hunk 1]
```diff
@@ -179,9 +179,8 @@
 #                                                                                                                ^ source.hlasmListing keyword.other.hlasm
 #                                                                                                                 ^^^^^^^^^^^^^^^^^^^^ source.hlasmListing invalid.ignored
 >                                                       REMARKS                                                  1234567
```
Comment (VitGottwald) [inline clients/vscode-hlasmplugin/src/test/syntaxes/endevor/basics.listing.snap:181]: Comment should start in column 16. The spaces before are not part of the comment. Not sure if it is worth fixing if that would be complicated.

File: clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt [hunk 2]
```diff
@@ -400,8 +400,12 @@
       }
     },
     "remark": {
-      "match": "(\\S+)(?<=^${listingDetails}$.{0,71})",
-      "name": "comment.hlasm"
+      "match": "\\s*(\\S.*)(?<=^${listingDetails}$.{0,71})",
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt:403]: This rule matches an empty string...

Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt:403]: maybe
```suggestion
      "match": "(\\S.*)(?<=^${listingDetails}$.{0,71})",
```
?

File: clients/vscode-hlasmplugin/src/test/syntaxes/endevor/basics.listing.snap (hunk 1 above)
Comment (jirimosinger) [reply clients/vscode-hlasmplugin/src/test/syntaxes/endevor/basics.listing.snap:181]: I've adjusted it in a way that multiline remarks always start at column 16 regardless if there are just spaces on the line. Was this the goal?

File: clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt (hunk 2 above)
Comment (jirimosinger) [reply clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt:403]: I needed to also include `${listingDetails}$` but yeah 👍

File: clients/vscode-hlasmplugin/src/test/syntaxes/endevor/basics.listing.snap (hunk 1 above)
Comment (VitGottwald) [reply clients/vscode-hlasmplugin/src/test/syntaxes/endevor/basics.listing.snap:181]: Yes, perfect. Thank you!

Comment (VitGottwald) [reply clients/vscode-hlasmplugin/src/test/syntaxes/endevor/basics.listing.snap:181]: One more comment. Looking at line 178
```
#                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing
```
This should technically be two
```
#                                                            ^ source.hlasmListing
#                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing comment.hlasm
```

File: clients/vscode-hlasmplugin/src/test/syntaxes/endevor/mix.listing.snap [hunk 3]
```diff
@@ -641,19 +641,18 @@
 #^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing
 #                                                        ^^^ source.hlasmListing punctuation.definition.tag.hlasm
 #                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing
-#                                                                                    ^^^^^^^^^^^^^^^^ source.hlasmListing comment.hlasm
-#                                                                                                    ^^^^^^^^^^^^ source.hlasmListing
+#                                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing comment.hlasm
```
Comment (VitGottwald) [inline clients/vscode-hlasmplugin/src/test/syntaxes/endevor/mix.listing.snap:644]: Strictly speaking, the comment starts right after the space after the `OP2` operannd.

File: clients/vscode-hlasmplugin/src/test/syntaxes/endevor/mix.listing.snap [hunk 4]
```diff
@@ -721,8 +700,8 @@
 #                                                          ^^ source.hlasmListing
 #                                                            ^^ source.hlasmListing punctuation.definition.tag.hlasm variable.other.readwrite.hlasm
 #                                                              ^^^^^^^^^^^^^^^^^^^^ source.hlasmListing
-#                                                                                  ^^^^^^ source.hlasmListing comment.hlasm
-#                                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing
+#                                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing comment.hlasm
```
Comment (VitGottwald) [inline clients/vscode-hlasmplugin/src/test/syntaxes/endevor/mix.listing.snap:703]: Again, the comment starts right after `*-&P `.

File: clients/vscode-hlasmplugin/src/test/syntaxes/endevor/parenthesis.listing.snap [hunk 5]
```diff
@@ -144,8 +144,8 @@
 #                                                           ^ source.hlasmListing
 #                                                            ^^^ source.hlasmListing punctuation.definition.tag.hlasm
 #                                                               ^^^^^^^^^^^^^^ source.hlasmListing
-#                                                                             ^^^^^^^^^^^ source.hlasmListing comment.hlasm
-#                                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing
+#                                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing comment.hlasm
```
Comment (VitGottwald) [inline clients/vscode-hlasmplugin/src/test/syntaxes/endevor/parenthesis.listing.snap:147]: Same comment here, the comment starts after `OP1,OP2 `

File: clients/vscode-hlasmplugin/src/test/syntaxes/endevor/parenthesis.listing.snap [hunk 6]
```diff
@@ -163,8 +163,8 @@
 #                                                                       ^^^ source.hlasmListing punctuation.definition.tag.hlasm
 #                                                                          ^ source.hlasmListing
 #                                                                           ^^ source.hlasmListing
```
Comment (VitGottwald) [inline clients/vscode-hlasmplugin/src/test/syntaxes/endevor/parenthesis.listing.snap:165]: Same thing. The (single) space before `REMARK(REM)` should be part of the comment.

File: clients/vscode-hlasmplugin/src/test/syntaxes/endevorLong/basics.listing.snap [hunk 7]
```diff
@@ -287,20 +286,18 @@
 #                                                                 ^ source.hlasmListing
 #                                                                  ^^^ source.hlasmListing punctuation.definition.tag.hlasm
 #                                                                     ^^^^ source.hlasmListing
-#                                                                         ^^^^^^^ source.hlasmListing comment.hlasm
-#                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing
+#                                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing comment.hlasm
```
Comment (VitGottwald) [inline clients/vscode-hlasmplugin/src/test/syntaxes/endevorLong/basics.listing.snap:289]: Again, spaces before `REMARKS` (except the one right after `OP1,OP2`) should be part of the comment.

File: clients/vscode-hlasmplugin/src/test/syntaxes/endevorLong/parenthesis.listing.snap [hunk 8]
```diff
@@ -142,8 +142,8 @@
 #                                                                    ^ source.hlasmListing
 #                                                                     ^^^ source.hlasmListing punctuation.definition.tag.hlasm
 #                                                                        ^^^^^^^^^^^^^^ source.hlasmListing
-#                                                                                      ^^^^^^^^^^^ source.hlasmListing comment.hlasm
-#                                                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing
+#                                                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing comment.hlasm
```
Comment (VitGottwald) [inline clients/vscode-hlasmplugin/src/test/syntaxes/endevorLong/parenthesis.listing.snap:145]: Same thing. Spaces before `REMARK(REM)`, except the one right after `OP1,OP2` should be part of the comment.

File: clients/vscode-hlasmplugin/src/test/syntaxes/general/parenthesis.listing.snap [hunk 9]
```diff
@@ -141,8 +141,8 @@
 #                                                          ^ source.hlasmListing
 #                                                           ^^^ source.hlasmListing punctuation.definition.tag.hlasm
 #                                                              ^^^^^^^^^^^^^^ source.hlasmListing
-#                                                                            ^^^^^^^^^^^ source.hlasmListing comment.hlasm
-#                                                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing
+#                                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hlasmListing comment.hlasm
```
Comment (VitGottwald) [inline clients/vscode-hlasmplugin/src/test/syntaxes/general/parenthesis.listing.snap:144]: Same thing - spaces before `REMARK(REM)`

**Review by VitGottwald — APPROVED** [review]
Very nice. This removes a lot of noise from the snapshots. Thank you @jirimosinger! Feel free to address my comments in a separate PR.

**Review by VitGottwald — APPROVED** [review]
Very nice. This removes a lot of noise from the snapshots. Thank you @jirimosinger! Feel free to address my comments in a separate PR.

File: clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt [hunk 10]
```diff
@@ -400,8 +400,20 @@
       }
     },
     "remark": {
-      "match": "(\\S+)(?<=^${listingDetails}$.{0,71})",
-      "name": "comment.hlasm"
+      "patterns": [
+        {
+          "match": "^${listingDetails}$\\s{15}(.*)(?<=^${listingDetails}$.{0,71})",
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt:405]: Doesn't this interfere with the simple operand continuation?

**Review by slavek-kucera — APPROVED** [review]

File: clients/vscode-hlasmplugin/src/test/syntaxes/endevor/basics.listing.snap (hunk 1 above)
Comment (jirimosinger) [reply clients/vscode-hlasmplugin/src/test/syntaxes/endevor/basics.listing.snap:181]: Alright, adjusted 👍

File: clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt (hunk 10 above)
Comment (jirimosinger) [reply clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt:405]: Do you mean this

```
LABEL INSTR OP1,OP2,                                                   -12346578
               OP3
```


or this

```
LABEL INSTR                                                        OPER-12346578
               AND
```
?


The rule is now (slightly) changes, however, in either case, it looks fine:
image


However, this specific test case seems to be missing so I might add it later

**Pull request merged**